### PR TITLE
Fix reference ordering and formating

### PR DESCRIPTION
**Pull request by fmahebert**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [x] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsvu — APPROVED** [review]
Thanks @fmahebert!

File: docs/References.bib [hunk 1]
```diff
@@ -128,22 +128,22 @@ @article{Fishbone1976apj
 }
 
 @article{Foucart2015vpa,
```
Comment (kidder) [inline docs/References.bib:130]: and what determines the formatting?  If I copy from inspire, I get the formatting you are changing...

Comment (fmahebert) [reply docs/References.bib:130]: Consistency with the majority of the file, where keys are indented by two spaces, and the equal signs are aligned two spaces beyond the longest key.

Comment (kidder) [reply docs/References.bib:130]: well to be honest, this is not something I'll be doing when adding references, so if there is a desire to unify the format there will need to be a tool to do so...

Comment (fmahebert) [reply docs/References.bib:130]: @kidder I definitely agree, and we already have an issue for this: #1381.

**Review by nilsdeppe — APPROVED** [review]
LGTM

**Pull request merged**